src/pages/category/Category.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useRef, useEffect } from 'react'
+import { connect } from 'react-redux'
+import { nanoid } from 'nanoid'
+import { Card, Button, Table, Modal, Form, Input, message } from 'antd';
+import { PlusCircleOutlined } from '@ant-design/icons';
+import { reqCategoryList } from '../../api/index'
+import { pageSize } from '../../config'
+import { saveCategoryListAction } from '../../redux/actions/category'
+
+function Category(props) {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [operType, setOperType] = useState('') //为了复用modal框
+  const [categoryList, setCategoryList] = useState([])
+  const [isLoading, setIsloading] = useState(true) //Loading of table
+  const [modalCurrentValue, setModalCurrentValue] = useState({})
+  const formInstance = useRef()
+
+
+
+  useEffect(() => {
+    const getCategoryList = async () => {
+      let result = await reqCategoryList()
+      let { data } = result
+      props.saveCategoryList(data)  //save redux
+      let finalData = data.map((item, index) => {
+        return { "key": index, "categoryName": item }
+      })
+      setCategoryList(finalData)
+      setIsloading(false)
+    }
+    getCategoryList()
+  }, [])  //eslint-disable-line
+
+  const showAdd = () => {
+    setModalCurrentValue({})
+    setOperType('Add category')
+    setIsModalVisible(true);
+  };
+  const showUpdate = (e) => {
+    const { key, categoryName } = e
+    setModalCurrentValue({ key, categoryName })
+    setOperType('Update category')
+    setIsModalVisible(true);
+  };
+
+  const handleOk = (value) => {
+    //表单统一验证
+    formInstance.current.validateFields()
+      .then((res) => {
+        const { categoryName } = res
+        if (operType === 'Add category') {
+          //由于没有添加分类API，所以并未更新数据库
+          const newCategoryList = [{ "key": nanoid(), categoryName }]
+          setCategoryList([...newCategoryList, ...categoryList])
+          message.success('Added successfully!', 1)
+        }
+        if (operType === 'Update category') {
+          //问题：由于没有更新分类API，所以无法实现更新分类名(更新状态里的对象属性值，React没有重新渲染)
```

Comment:
React only do shallow comparison for state updating, you need to change the object reference
For better coding experience, suggest to use https://github.com/immerjs/immer